wgsl/index.bs

```diff
@@ -3060,15 +3090,82 @@ Issue:
 #### Input-output Locations TODO #### {#input-output-locations}
 TODO: *Stub*. Location-sizing of types, non-overlap among variables referenced within an entry point static call tree.
 
-### Resource interface TODO ### {#resource-interface}
+### Resource interface ### {#resource-interface}
+
+A <dfn noexport>resource</dfn> is an object,
+other than a [[#pipeline-inputs-outputs|pipeline input or output]],
+which provides access to data external to a [=shader stage=].
+Resources are shared by all invocations of the shader.
+
+There are four kinds of resources:
+
+* [=uniform buffers=]
+* [=storage buffers=]
+* textures
+* samplers
+
+The <dfn noexport>resource interface of a shader</dfn> is the set of module-scope
+resource variables [=statically accessed=] by
+[=functions in a shader stage|functions in the shader stage=].
+
+Each resource variable must be declared with both <dfn noexport>`set`</dfn>
+and <dfn noexport>`binding`</dfn> attributes.
+Together with the shader's stage, these identify the binding address
+of the resource on the shader's pipeline.
+See [[WebGPU#pipeline-layout|WebGPU &sect; GPUPipelineLayout]].
+
+Bindings must not alias within a shader stage:
+two different variables in the resource interface of a given
+shader must not have the same set and binding values, when considered as a pair of values.
```

Comment:
I have to check this no-overlap-bindings rule for samplers and their associated sampled textures.

Reply:
I think the only cases were:
- buffer resources that alias the same memory.  But I believe that won't be allowed in WebGPU
- Vulkan has combined image+sampler, and you can make OpTypeSampler, OpTypeImage, and OpTypeSampledImage variables to reference that descriptor, all with the same set and binding.  But I believe WebGPU won't have that.

So I think this i the right rule. Yay.